cmd/dra-example-controller/driver.go

```diff
@@ -244,18 +246,19 @@ func (d driver) unsuitableNode(ctx context.Context, pod *corev1.Pod, allcas []*c
 
 	perKindCas := make(map[string][]*controller.ClaimAllocation)
 	for _, ca := range allcas {
-		var kind string
-		switch ca.ClaimParameters.(type) {
-		case *gpucrd.GpuClaimParametersSpec:
-			kind = gpucrd.GpuClaimParametersKind
+		switch ca.Claim.Spec.ParametersRef.Kind {
+		case gpucrd.GpuClaimParametersKind:
+			perKindCas[gpucrd.GpuClaimParametersKind] = append(perKindCas[gpucrd.GpuClaimParametersKind], ca)
+		default:
+			return fmt.Errorf("unsupported ResourceClaimParameters kind: %+v", ca.Claim.Spec.ParametersRef.Kind)
 		}
-		perKindCas[kind] = append(perKindCas[kind], ca)
 	}
 	for _, kind := range []string{gpucrd.GpuClaimParametersKind} {
 		var err error
 		switch kind {
 		case gpucrd.GpuClaimParametersKind:
 			err = d.gpu.UnsuitableNode(crd, pod, perKindCas[kind], allcas, potentialNode)
+		default:
```

Comment:
Returning an error missing here?

Reply:
We're looping through the static list of items, the only reason the default is here is because linter would request replacing this with if. I think returning an error in here is a bit too self-doubting. because the only error that can possibly take place is somebody extending the list to range-loop through but without adding respective case in switch.

Reply:
I understand that this is a matter of taste mostly, but I'd agree with linter here.
This
```
if ca.Claim.Spec.ParametersRef.Kind != ca.Claim.Spec.ParametersRef.Kind {
        return fmt.Errorf("unsupported ResourceClaimParameters kind: %+v", ca.Claim.Spec.ParametersRef.Kind)
}
perKindCas[gpucrd.GpuClaimParametersKind] = append(perKindCas[gpucrd.GpuClaimParametersKind], ca)
```
is much more readable in my opinion.
However, I understand that it doesn't suggest that there *might* be other kinds here.

Reply:
> because the only error that can possibly take place is somebody extending the list to range-loop through but without adding respective case in switch

And that is exactly what the code should be checking for.

Reply:
> However, I understand that it doesn't suggest that there might be other kinds here.

Documenting that is the purpose of using `switch` instead of `if`.

Reply:
> And that is exactly what the code should be checking for.

I'd say this is a job for unit tests, The code base would explode if we tried to cover all places where the developer can shoot self in the foot.

Reply:
Forgot this, had to force-push once more, sorry.
Done.

Reply:
> Documenting that is the purpose of using switch instead of if.

Yep. The question is does it worth it. And the answer is very much a matter of taste in my opinon. My answer is no, just in case it's not yet clear :)